fastembed/embedding.py

```diff
@@ -393,23 +419,25 @@ def retrieve_model_gcs(self, model_name: str, cache_dir: str) -> Path:
 
         return model_dir
 
+    @gcs_fallback
```

Comment:
Not a fan of this decorator approach. Makes the code less obvious/skimmable. Would prefer a more explicit fallback of some sort. 

Additionally, should call it `web_url_download` or something similar. Expose the url to the user as well. E.g. if the model source is a URL, use that instead.

Reply:
Preferred the decorator since the other options were nested try/except or conditionals. The decorator enables us to unplug it very easily when we drop support for GCS.